interface/frontend/src/components/open-image/OpenDICOM.vue

```diff
@@ -0,0 +1,133 @@
+<template>
+  <div class="DICOM-container">
+    <div class="DICOM" ref="DICOM"></div>
+  </div>
+</template>
+
+<script>
+  import { EventBus } from '../../main.js'
+  const cornerstone = require('cornerstone-core')
+  const Q = require('q')
```

Comment:
`q` is not very well maintained these day and it is also very old. I would suggest either looking forward with the Promise API implemented within the browser itself: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise

Or use BlueBird for the perf boost as well as the issue discussed in this article: https://alexn.org/blog/2017/10/11/javascript-promise-leaks-memory.html

Blue bird: http://bluebirdjs.com/docs/getting-started.html

Reply:
Good point, done by [this](https://github.com/concept-to-clinic/concept-to-clinic/pull/188/commits/f8347df03addd788ca16224f3db246d20c7ad93c).